spec/02-integration/05-proxy/18-upstream_tls_spec.lua

```diff
@@ -125,26 +129,93 @@ for _, strategy in helpers.each_strategy() do
         paths = { "/mtls-upstream", },
       }))
 
+      -- tls
+      tls_service_mtls = assert(bp.services:insert({
+        name = "tls-protected-service-mtls",
+        url = "tls://127.0.0.1:16798",
+      }))
+
+      tls_service_tls = assert(bp.services:insert({
+        name = "tls-protected-service",
+        url = "tls://example.com:16799", -- domain name needed for hostname check
+      }))
+
+      tls_upstream = assert(bp.upstreams:insert({
+        name = "tls-backend-mtls",
+      }))
+
+      assert(bp.targets:insert({
+        upstream = { id = tls_upstream.id, },
+        target = "example.com:16798",
+      }))
+
+      tls_service_mtls_upstream = assert(bp.services:insert({
+        name = "tls-protected-service-mtls-upstream",
+        url = "tls://tls-backend-mtls",
+        host = "example.com"
+      }))
+
+      assert(bp.routes:insert({
+        service = { id = tls_service_mtls.id, },
+        destinations = {
+          {
+            port = 19000,
+          },
+        },
+        protocols = {
+          "tls",
+        },
+      }))
+
+      assert(bp.routes:insert({
+        service = { id = tls_service_tls.id, },
+        destinations = {
+          {
+            port = 19001,
+          },
+        },
+        protocols = {
+          "tls",
+        },
+      }))
+
+      assert(bp.routes:insert({
+        service = { id = tls_service_mtls_upstream.id, },
+        destinations = {
+          {
+            port = 19002,
+          },
+        },
+        protocols = {
+          "tls",
+        },
+      }))
+
+
       assert(helpers.start_kong({
         database   = strategy,
         nginx_conf = "spec/fixtures/custom_nginx.template",
+        stream_listen = helpers.get_proxy_ip(false) .. ":19000," ..
+        helpers.get_proxy_ip(false) .. ":19001," ..
+        helpers.get_proxy_ip(false) .. ":19002"
```

Comment:
A little bit of indentation change for readability + adding a trailing comma in the table:

```suggestion
        stream_listen = helpers.get_proxy_ip(false) .. ":19000,"
                     .. helpers.get_proxy_ip(false) .. ":19001,"
                     .. helpers.get_proxy_ip(false) .. ":19002",
```